### DISCUSSION TITLE
I can´t run a live timing recorded file

### DISCUSSION THREAD

**Discussion by ChemiF1** [theOehrly/Fast-F1#648] [Q&A]
Hello everyone

During the last weeks i tried to run a live timing recorded file using the instructions you gave on your site (https://docs.fastf1.dev/livetiming.html)

I got the txt file without any problem, (here it is [austin_sprintrace.txt](https://github.com/user-attachments/files/17507144/austin_sprintrace.txt)) but after several attemps but i wasn´t able to load the data. 

The code i used and the message the system gave to me was the following one

![image](https://github.com/user-attachments/assets/b17478e4-9209-48f9-aa03-afd82976c4f5)

Anyone could indicate to me what am i doing wrong? 

Thank you so much

**Comment by theOehrly** [comment]
I've figured out what's the problem here, mostly. The data isn't quite as expected. Not sure why that is. I'll need to do some further research to maybe find that out.

For now, there's nothing you can do to fix this.

Have you noted this issue with multiple sessions or is it specific to only this one session?

**Comment by ChemiF1** [comment]
Finally i was able to run properly this, without major problems.

The file i recorded was this 
[qatar_race.txt](https://github.com/user-attachments/files/18012515/qatar_race.txt)

Some quick questions arose from there

1. Can you check the file is correct?
2. Which Python version do i have to set to record and load the data? Because i tried to do in 3.12 and i was able to get it (fastf1 version 3.4.3)
3. How much time do i have to wait after session ends to finish the recording?
4. Can I load the load the recorded data just after the recording finishes? Or do I have to wait for something else?

Thank you for your work, magician!

**Reply by theOehrly** [reply]
>     2. Which Python version do i have to set to record and load the data? Because i tried to do in 3.12 and i was able to get it (fastf1 version 3.4.3)

Python version doesn't matter (out of the ones that are supported by Fastf1, that's currently 3.9-3.13)

>     3. How much time do i have to wait after session ends to finish the recording?

I always just let it run until it times out by itself. You might miss out on a few more or less important race control messages if you end it manually right after the last lap.

>     4. Can I load the load the recorded data just after the recording finishes? Or do I have to wait for something else?

Yes, you can. The only things that's missing then is data from Ergast. But that is also missing for non-recorded sessions because Ergast is slower to update.

>     1. Can you check the file is correct?

It looks mostly correct. But the recording didn't start early enough. It doesn't capture the laps to the grid. Therefore, FastF1 believes that the cars aren't on track yet until the first pit stop of each driver. For example, Hamilton has only 23 laps which is the exact number of laps after his first pit stop. So you need to start the recording earlier. I have no idea how much earlier. I have never tried to record a race session myself 😅

I don't think this info is in the documentation. I'll try to add it soon.

By the way, why do you want to go to the trouble of recording the sessions? Why don't you just use the archived data?

**Reply by ChemiF1** [reply]
I was able to keep learning about the system during the last race and i got some key points. Among them I was able to get the loading process, how to record and how not to record, which Python package i needed and some stuff about that. 

The main one was i only could get the full data about 25min later the session had ended. Before of that i got this situation

![image](https://github.com/user-attachments/assets/609771e8-bfca-4dfd-90e7-686231ea0952)
![image](https://github.com/user-attachments/assets/bff38552-e5c4-4bda-874d-a5f27abe2373)
![image](https://github.com/user-attachments/assets/3ba4303c-d856-430d-9eb9-e5dde90d58e4)

And about 25min later i was able to get the full data and i can get the full data of the session

![image](https://github.com/user-attachments/assets/c20d4aa1-8145-4d74-ace5-798a4a6ae2ed)
![image](https://github.com/user-attachments/assets/5b76f070-5833-4b3e-91fb-102cf55e3f86)

There were something missing there, but from you´re saying it´s very probably i can´t get the full data before Ergast has ended the session info. So, is it Ergast related? Can I do something to speed up the process?

> By the way, why do you want to go to the trouble of recording the sessions? Why don't you just use the archived data?

The reason i´m going through this way is to be able to get the session data as soon as possible to not have to wait in the analysis process. I´m helping some friends with their project (you can see [here](https://x.com/DPerformanceF1)) and time really counts very much after FP1 and FP3.

**Reply by theOehrly** [reply]
Answer below, I failed to respond to this discussion thread 😅

**Comment by theOehrly** [comment]
This turns out to be somewhat complicated. Looking at the first log, FastF1 fails to load "Session Info", "Session Status", "Timing Data" and "Result Data from Ergast".

- "Session Info" is not contained in your recording. I'm not entirely sure, but I think that you'd need to start recording earlier, maybe. This is metadata about the session, which may only be provided once at the beginning of the session. I'll try to investigate this but I'm not sure I can do anything about it before next season.

- "Session Status" is missing from the recording as well, but I'm pretty sure this is a bug in the data recorder.

- "Timing Data" is errors during data loading because it requires a value from session info. 

Despite providing the recorded data, FastF1 tries to fall back to the API if it is unable to get data from the recording. This fixes the problem with missing session info data (about 25 minutes later, as you noted). Also, loading timing data now succeeds because of that.

The data from Ergast is used to populate the lap time for the very first lap and the results (``Session.results`` as well as the data in ``Session.get_driver()``). Therefore, you can't get the full data before Ergast was updated (which may take hours), but you can get most of the data.

> The main one was i only could get the full data about 25min later the session had ended. Before of that i got this situation
